### PR TITLE
[NavigationDrawer] Allow header to expand even if content doesn't fill screen.

### LINKED ISSUE #7684: [NavigationDrawer] Internal issue: b/135759737
This was filed as an internal issue. If you are a Googler, please visit [b/135759737](http://b/135759737) for more details.

---
#### Internal data
- Associated internal bug: [b/135759737](http://b/135759737)

### PR DESCRIPTION
**Pull request by codeman7**
Previously the MDCBottomDrawerViewController would not expand the header if the content did not reach full screen. This change makes it so that clients can have the header always expand the header regardless of the total content height. This new behavior is behind a behavioral flag so that clients can opt into this if they would prefer that. This change is so that clients can still have their content visible when the header goes into the safe area.

## Screenshots
_iPhone XS Max_

| Before | After |
| - | - |
|beforeXSMax|largeDeviceAfter|

_iPhone 4s_

| Before | After |
| - | - |
|4sbefore|4safter|

Closes #7684

### CODE REVIEW

**Review by randallli — APPROVED** [review]

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m [hunk 1]
```diff
@@ -643,6 +643,21 @@ - (void)updateViewWithContentOffset:(CGPoint)contentOffset {
   CGFloat fullscreenHeaderHeight =
       self.contentReachesFullscreen ? self.topHeaderHeight : [self contentHeaderHeight];
 
+  CGFloat contentHeight = self.contentViewController.preferredContentSize.height +
```
Comment (yarneo) [inline components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m]: calling the preferredContentSize of the contentViewController every contentOffset change of the scroll will cause degraded performance in clients with a calculated (more complex) preferredContentSize for their view controller. This was seen with OneGoogle, hence the addition of _contentVCPreferredContentSizeHeightCached

Comment (yarneo) [reply components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m]: The same could possibly be said for the headerViewController preferredContentSize.

**Review by yarneo — CHANGES_REQUESTED** [review]

File: components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m (hunk 1 above)
Comment (codeman7) [reply components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m]: Addressed in [#04c8695](https://github.com/material-components/material-components-ios/pull/8484/commits/04c869573f5d35f2f0f18ae0aed189d0ad28f327)

**Review by yarneo — APPROVED** [review]
overall looks good. there seems to be additional calculations taking place now on top of all the existing ones. is there no existing property that already encapsulates these calculations?

Also, were you able to verify all the examples still work as intended for portrait and landscape, with and without a notch?

**Comment by codeman7** [comment]
@yarneo 
1. Using existing properties: I couldn't find any for these but I think we could do some refactoring to adjust that in the future. Let's chat offline about how we want to approach this.
2. Examples: I've verified in both landscape and portrait the examples WAI on both a 4s and XS Max

**Comment by yarneo** [comment]
Sounds good, thanks

**Pull request merged**